src/components/CategoryFilter.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { filter } from '../actions/index';
+
+const CategoryFilter = props => {
+  const { filterOptions, handleFilterChange } = props;
+
+  const optionList = () => (
+    filterOptions.map(option => (
+      <option key={`${option}-key`} value={option}>{option}</option>
+    ))
+  );
+
+  const handleEventFilterChange = event => {
+    handleFilterChange(event.target.value);
+  };
+
+  return (
+    <label htmlFor="filter" className="filter-style">
+      Filter:
+      <select id="filter" onChange={handleEventFilterChange}>
+        {optionList()}
+      </select>
+    </label>
+  );
+};
+
+CategoryFilter.propTypes = {
+  filterOptions: PropTypes.arrayOf(PropTypes.string).isRequired,
+  handleFilterChange: PropTypes.func.isRequired,
+};
+
+const mapDispatchToProps = dispatch => ({
+  handleFilterChange: label => {
+    dispatch(filter(label));
+  },
+});
+
+const ConnectedCategoryFilter = connect(null, mapDispatchToProps)(CategoryFilter);
```

Comment:
**Required**: Required:CategoryFilter should be a presentational component, it should not connect to the Redux store.

`handleFilterChange` should be on the `BookList` component and you should pass it down to this component.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/HandelChangeFilter.png)